operations/consumergroupoffsets/consumer-group-offset-operation.go

```diff
@@ -0,0 +1,217 @@
+
+	// admin.ListConsumerGroupOffsets(group, nil) can be used to fetch the offsets when
+	// https://github.com/Shopify/sarama/pull/1374 is merged
+	coordinator, err := consumer.client.Coordinator(consumer.groupName)
+	if err != nil {
+		output.Failf("failed to get coordinator: %v", err)
+	}
+
+	request := &sarama.OffsetFetchRequest{
+		// this will only work starting from version 0.10.2.0
+		Version:       2,
+		ConsumerGroup: consumer.groupName,
+	}
+
+	groupOffsets, err := coordinator.FetchOffset(request)
+	if err != nil {
+		output.Failf("failed to get fetch group offsets: %v", err)
+	}
+
+	offsets := make([]partitionOffsets, 0)
+
+	if flags.Partition > -1 {
+		offset := resetOffset(consumer.client, flags.Partition, flags, groupOffsets, session)
+		offsets = append(offsets, offset)
+	} else {
+
+		partitions, err := consumer.client.Partitions(flags.Topic)
+		if err != nil {
+			output.Failf("failed to list partitions: %v", err)
+		}
+
+		for _, partition := range partitions {
+			offset := resetOffset(consumer.client, partition, flags, groupOffsets, session)
+			offsets = append(offsets, offset)
+		}
+	}
+
+	if flags.OutputFormat != "" {
+		output.PrintObject(offsets, flags.OutputFormat)
+	} else {
+		tableWriter := output.CreateTableWriter()
+		tableWriter.WriteHeader("PARTITION", "OLDEST_OFFSET", "NEWEST_OFFSET", "CURRENT_OFFSET", "TARGET_OFFSET")
+		for _, o := range offsets {
+			tableWriter.Write(strconv.FormatInt(int64(o.Partition), 10),
+				strconv.FormatInt(int64(o.OldestOffset), 10), strconv.FormatInt(int64(o.NewestOffset), 10),
+				strconv.FormatInt(int64(o.CurrentOffset), 10), strconv.FormatInt(int64(o.TargetOffset), 10))
+		}
+		tableWriter.Flush()
+	}
+	return nil
+}
+
+func (consumer *Consumer) Cleanup(sarama.ConsumerGroupSession) error {
+	close(consumer.ready)
+	return nil
+}
+
+func (consumer *Consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	return nil
+}
+
+func getPartitionOffsets(client sarama.Client, partition int32, flags ResetConsumerGroupOffsetFlags) partitionOffsets {
+
+	var err error
+	offsets := partitionOffsets{Partition: partition}
+
+	if offsets.OldestOffset, err = client.GetOffset(flags.Topic, partition, sarama.OffsetOldest); err != nil {
+		output.Failf("failed to get offset for topic %s Partition %d: %v", flags.Topic, partition, err)
+	}
+
+	if offsets.NewestOffset, err = client.GetOffset(flags.Topic, partition, sarama.OffsetNewest); err != nil {
+		output.Failf("failed to get offset for topic %s Partition %d: %v", flags.Topic, partition, err)
+	}
+
+	if flags.Offset > -1 {
+		if flags.Offset < offsets.OldestOffset {
+			output.Failf("cannot set offset for Partition %d: offset (%d) < oldest offset (%d)", partition, flags.Offset, flags.OldestOffset)
+		} else if flags.Offset > offsets.NewestOffset {
+			output.Failf("cannot set offset for Partition %d: offset (%d) > newest offset (%d)", partition, flags.Offset, flags.NewestOffset)
```

Comment:
These two error messages seem a bit off. It should probably be:
```suggestion
			output.Failf("cannot set offset for Partition %d: offset (%d) > newest offset (%d)", partition, flags.Offset, offsets.NewestOffset)
```